app/code/Bulbulatory/Recommendations/Model/RecommendationRepository.php

```diff
@@ -20,32 +25,154 @@ class RecommendationRepository implements RecommendationRepositoryInterface
      */
     private $recommendationCollectionFactory;
 
+    /**
+     * @var Random
+     */
+    private $mathRandom;
+
+    /**
+     * RecommendationRepository constructor.
+     * @param RecommendationFactory $recommendationFactory
+     * @param RecommendationCollectionFactory $recommendationCollectionFactory
+     * @param Random $mathRandom
+     */
     public function __construct(
         RecommendationFactory $recommendationFactory,
-        RecommendationCollectionFactory $recommendationCollectionFactory
+        RecommendationCollectionFactory $recommendationCollectionFactory,
+        Random $mathRandom
     ) {
         $this->recommendationFactory = $recommendationFactory;
         $this->recommendationCollectionFactory = $recommendationCollectionFactory;
+        $this->mathRandom = $mathRandom;
     }
 
+    /**
+     * @param int $id
+     * @return Recommendation
+     */
     public function getById($id)
     {
         $recommendation = $this->recommendationFactory->create();
         $recommendation->getResource()->load($recommendation, $id);
+
         if (! $recommendation->getId()) {
             throw new NoSuchEntityException(__('Unable to find recommendation with ID "%1"', $id));
         }
+
+        return $recommendation;
+    }
+
+    /**
+     * @param string $email
+     * @return mixed
+     */
+    private function getByEmail(string $email)
+    {
+        $recommendation = $this->recommendationFactory->create();
+        $recommendation->load($email, 'email');
+
+        if (! $recommendation->getId()) {
+            throw new NoSuchEntityException(__('Unable to find recommendation with email "%1"', $email));
+        }
+
         return $recommendation;
     }
 
+    /**
+     * @param string $hash
+     * @return mixed
+     */
+    private function getByHash(string $hash)
+    {
+        $recommendation = $this->recommendationFactory->create();
+        $recommendation->load($hash, 'hash');
+
+        if (! $recommendation->getId()) {
+            throw new NoSuchEntityException(__('Unable to find recommendation with hash "%1"', $hash));
+        }
+
+        return $recommendation;
+    }
+
+    /**
+     * @param Recommendation $recommendation
+     * @return Recommendation
+     */
     public function save(Recommendation $recommendation)
     {
         $recommendation->getResource()->save($recommendation);
         return $recommendation;
     }
 
+    /**
+     * @param Recommendation $recommendation
+     */
     public function delete(Recommendation $recommendation)
     {
         $recommendation->getResource()->delete($recommendation);
     }
+
+    /**
+     * @param array $data
+     * @return int
+     * @throws Exception
+     * @throws Throwable
+     */
+    public function create(array $data)
+    {
+        // Validate data for create Recommendation
+        if(empty($data['customer_id'])) throw new Exception(__('Customer Id is empty!'));

```

Comment:
Lets keep the brackets, it's more readable.
```suggestion
        if(empty($data['customer_id'])) {
            throw new Exception(__('Customer Id is empty!'));
        }
```

Reply:
@TM18

> Lets keep the brackets, it's more readable.

Fixed!